uint128_test.go

```diff
@@ -399,3 +440,77 @@ func BenchmarkString(b *testing.B) {
 		}
 	})
 }
+
+// raw 128 bits
+type b128 [128]bool
```

Comment:
this is...an interesting approach.

Since `big.Int` doesn't support all the bitwise operations we need, I think it's fine to test a few cases manually instead. We don't need exhaustive random testing here.